xtext/org.icyphy.linguafranca/src/org/icyphy/Target.java

```diff
@@ -347,104 +361,236 @@
+                    for (KeyValuePair entry : value.getKeyvalue().getPairs()) {
+                        // FIXME: convert to string values here, not later on
+                        // during code generation.
+                        config.clockSyncOptions.put(
+                                (ClockSyncOption) DictionaryType.CLOCK_SYNC_OPTION_DICT
+                                        .forName(entry.getName()),
+                                entry.getValue());
+                    }
+                }),
+        
         /**
-         * Directive to specify a cmake to be included by the generated build systems.
+         * Directive to specify a cmake to be included by the generated build
+         * systems.
          *
-         * This gives full control over the C++ build as any cmake parameters can be adjusted in the included file.
+         * This gives full control over the C++ build as any cmake parameters
+         * can be adjusted in the included file.
          */
-        CMAKE_INCLUDE("cmake-include", Arrays.asList(Target.CPP)),
+        CMAKE_INCLUDE("cmake-include", PrimitiveType.STRING,
+                Arrays.asList(Target.CPP), (config, value) -> {
+                    config.cmakeInclude = ASTUtils.toText(value);
+                }),
         
         /**
          * Directive to specify the target compiler.
          */
-        COMPILER("compiler", Arrays.asList(Target.ALL)),
+        COMPILER("compiler", PrimitiveType.STRING, Arrays.asList(Target.ALL),
+                (config, value) -> {
+                    config.compiler = ASTUtils.toText(value);
+                }),
         
         /**
          * Directive to let the execution engine allow logical time to elapse
          * faster than physical time.
          */
-        FAST("fast", Arrays.asList(Target.ALL)),
+        FAST("fast", PrimitiveType.BOOLEAN, Arrays.asList(Target.ALL),
+                (config, value) -> {
+                    config.fastMode = ASTUtils.toBoolean(value);
+                }),
         
         /**
          * Directive to stage particular files on the class path to be
          * processed by the code generator.
          */
-        FILES("files", Arrays.asList(Target.ALL)),
+        FILES("files", UnionType.FILE_OR_FILE_ARRAY, Arrays.asList(Target.ALL),
+                (config, value) -> {
+                    config.fileNames = ASTUtils.toListOfStrings(value);
+                }),
         
         /**
          * Flags to be passed on to the target compiler.
          */
-        FLAGS("flags", Arrays.asList(Target.C, Target.CCpp)),
+        FLAGS("flags", PrimitiveType.STRING,
+                Arrays.asList(Target.C, Target.CCPP), (config, value) -> {
+                    config.compilerFlags.clear();
+                    String str = ASTUtils.toText(value);
+                    if (!str.isEmpty()) {
+                        // FIXME: this is unsafe because it doesn't account for
```

Comment:
@Soroosh129: this came from a merge with code that you recently added. Could you take a stab at addressing this FIXME?

Reply:
Sure! Should we make it an actual list like the files property?

Reply:
Ah, good idea. That would be much better. In that case, I'll take care of it.